src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1601,8 +1601,18 @@ private MemberDeclarationSyntax ParseTypeDeclaration(SyntaxListBuilder<Attribute
 
                     return this.ParseClassOrStructOrInterfaceDeclaration(attributes, modifiers);
 
-                case SyntaxKind.RefKeyword:
                 case SyntaxKind.StructKeyword:
+                    // report use of "readonly struct" if feature is unsupported
+                    for (int i = 0, n = modifiers.Count; i < n; i++)
```

Comment:
>for (int i = 0, n = modifiers.Count; i < n; i++) [](start = 20, length = 48)

It feels like this for can be extracted into a helper method that takes modifier kind and feature id, and shared between case SyntaxKind.ClassKeyword and case SyntaxKind.StructKeyword.